Index: django/contrib/syndication/feeds.py
===================================================================
--- django/contrib/syndication/feeds.py	(revision 9811)
+++ django/contrib/syndication/feeds.py	(working copy)
@@ -30,8 +30,15 @@
         self.slug = slug
         self.request = request
         self.feed_url = request.path
-        self.title_template_name = self.title_template or ('feeds/%s_title.html' % slug)
-        self.description_template_name = self.description_template or ('feeds/%s_description.html' % slug)
+        self.appname = request.path.split('/')[1] # get app name from url
+    	if self.appname:
+    	    self.title_template_name = self.title_template or \
+    	      ('%s/feeds/%s_title.html' % (self.appname, slug))
+    	    self.description_template_name = self.description_template or \
+    	      ('%s/feeds/%s_description.html' % (self.appname, slug))
+	else:
+    	    self.title_template_name = self.title_template or ('feeds/%s_title.html' % slug)
+    	    self.description_template_name = self.description_template or ('feeds/%s_description.html' % slug)
 
     def item_link(self, item):
         try:
